pyinter/interval_set.py

```diff
@@ -84,6 +89,13 @@ def add(self, other):
         This has no effect if the Interval is already represented.
         :param other: an Interval to add to this IntervalSet.
         """
+        # If the other is a collection of intervals add them one by one.
+        try:
+            for inter in other:
+                self.add(inter)
+            return
+        except TypeError:
```

Comment:
I know, I don't like it either but as far as I am aware this is the only way to type check an iterable. If you want I can just change all of the call sites to add element by element.